datalad/core/local/diff.py

```diff
@@ -251,15 +254,12 @@ def _diff_ds(ds, fr, to, constant_refs, recursion_level, origpaths, untracked,
             untracked=untracked,
             eval_file_type=eval_file_type,
             _cache=cache)
-    except ValueError as e:
-        msg_tmpl = "reference '{}' invalid"
-        # not looking for a debug repr of the exception, just the message
-        estr = text_type(e)
-        if msg_tmpl.format(fr) in estr or msg_tmpl.format(to) in estr:
+    except InvalidGitReferenceError as e:
+        if e.ref in (fr, to):
```

Comment:
not even sure if this check is strictly needed here, but I guess would not hurt?

Reply:
I'd prefer not to have it, because we don't know what future `get_content_info()` implementations will do.